### PR TITLE
Implemented "explores, then it explores again"

### PR DESCRIPTION
**Pull request by acabaca24**
Before we could only get "it explores. It explores." or write the text ourselves

Now when creating the ability we can reference the number of explores done(guess that can be changed in the future, that's why it's a byte instead of a bool).

For more info see issue #6958

### CODE REVIEW

**Comment by JayDi85** [comment]
Original problem was in broken hint param (see fix example in 615fba5a72882f99f9aa1aacc34788ed01eec9eb).

File: Mage/src/main/java/mage/abilities/effects/keyword/ExploreSourceEffect.java [hunk 1]
```diff
@@ -49,6 +59,7 @@ public static String getRuleText(boolean showAbilityHint, String whosExplores) {
 
     private String sourceName = "it";
     private boolean showAbilityHint = true;
+    private byte numTimesExplored = 0;
```
Comment (JayDi85) [inline Mage/src/main/java/mage/abilities/effects/keyword/ExploreSourceEffect.java:62]: 1. You are added text about outside logic here. Ability contains info about times explored, but real effect logic still lay down in the parent ability as two added effects). That's wrong. If you add double times logic in effect then it must be here too (e.g. call `explorePermanent` multiple times in apply method).

2. Also it have non standard type: `byte`. Try to use standard types like `int` (if you try to find `byte` in xmage's code, can't find real usage of it).

**Review by JayDi85 — COMMENTED** [review]
Just for info

File: Mage/src/main/java/mage/abilities/effects/keyword/ExploreSourceEffect.java (hunk 1 above)
Comment (acabaca24) [reply Mage/src/main/java/mage/abilities/effects/keyword/ExploreSourceEffect.java:62]: Understandable! Thank you for the help!

**Pull request closed without merging**